drake/solvers/Optimization.h

```diff
@@ -0,0 +1,323 @@
+#ifndef DRAKE_OPTIMIZATION_H
+#define DRAKE_OPTIMIZATION_H
+
+#include "Core.h"
+#include <list>
+#include <memory>
+#include <initializer_list>
+
+namespace Drake {
+
+  struct DecisionVariable {
+    enum VarType {
+      CONTINUOUS,
+      INTEGER,
+      BINARY
+    };
+    VarType type;
+    std::string name;
+    Eigen::VectorXd value;
```

Comment:
I thought about this, but we still actually pass it as a continuous value to (and receive from) gurobi, etc.  and in fact we are only getting a numerical approximation of it back.